allennlp/training/metrics/spearman_correlation.py

```diff
@@ -58,17 +63,40 @@ def __call__(
             self.total_predictions = torch.cat((self.total_predictions, predictions), 0)
             self.total_gold_labels = torch.cat((self.total_gold_labels, gold_labels), 0)
 
+        if is_distributed():
+            try:
+                # The following logic will only work if the batches are of equal length.
+                world_size = dist.get_world_size()
+                device = gold_labels.device
+                _total_predictions = [
+                    torch.zeros(self.total_predictions.shape, device=device)
+                    for i in range(world_size)
+                ]
+                _total_gold_labels = [
+                    torch.zeros(self.total_gold_labels.shape, device=device)
+                    for i in range(world_size)
+                ]
+                dist.all_gather(_total_predictions, self.total_predictions)
+                dist.all_gather(_total_gold_labels, self.total_gold_labels)
+                self.total_predictions = torch.cat(_total_predictions, dim=0)
+                self.total_gold_labels = torch.cat(_total_gold_labels, dim=0)
+            except Exception:

```

Comment:
Can you catch something more specific? This will mask other errors too.

Reply:
Unfortunately, the exception that's thrown when the batches are unequal is of the generic type `Exception`.

Reply:
What I have done in the past in these cases is pattern match on the string form of the exception. This is a particularly egregious example: https://github.com/allenai/pipette/blob/master/pipette/pipette.py#L656

If we can't change the kind of exception that's thrown, this is a last resort. Catching everything is too dangerous. Even `MemoryError` is a type of `Exception`, and we definitely don't want to catch that and pretend it's handled.